app/controllers/form_controller.rb

```diff
@@ -8,8 +8,12 @@ def submit_form
       @page = @case_log.form.get_page(params[:case_log][:page])
       responses_for_page = responses_for_page(@page)
       if @case_log.update(responses_for_page) && @case_log.has_no_unresolved_soft_errors?
-        redirect_path = @case_log.form.next_page_redirect_path(@page, @case_log)
-        redirect_to(send(redirect_path, @case_log))
+        if @page.id == "declaration"
```

Comment:
Hmm, agree with the first point. But if we put it in the `next_page_redirect_path` we then use whatever is returned within `redirect_to(send(redirect_path, @case_log))` which sends us to /logs.#{log_id}, I assume because we pass `@case_log` through

Reply:
Is it okay to use ID in the model? we're using it `next_subsection` method already

Reply:
Ideally I think we should remove it there too